webapp/src/filelist.tsx

```diff
@@ -253,21 +254,102 @@ namespace custom {
         const targetTheme = pxt.appTarget.appTheme;
         const mainPkg = pkg.mainEditorPkg()
         const plus = show && !mainPkg.files[customFile]
-        const sync = show && pxt.github.token && !!mainPkg.header.githubId
+        const showGithub = show && (pxt.github.token || targetTheme.alwaysGithubItem);
         const meta: pkg.PackageMeta = this.getData("open-pkg-meta:" + mainPkg.getPkgId());
         return <div role="tree" className={`ui tiny vertical ${targetTheme.invertedMenu ? `inverted` : ''} menu filemenu landscape only hidefullscreen`}>
             <div role="treeitem" aria-selected={show} aria-expanded={show} aria-label={lf("File explorer toolbar")} key="projectheader" className="link item" onClick={this.toggleVisibility} tabIndex={0} onKeyDown={sui.fireClickOnEnter}>
                 {lf("Explorer")}
                 <sui.Icon icon={`chevron ${show ? "down" : "right"} icon`} />
-                {sync ? <sui.Button className="primary label" icon="github" title={lf("Sync with github")} onClick={this.handleSyncClick} onKeyDown={this.handleButtonKeydown} /> : undefined}
                 {plus ? <sui.Button className="primary label" icon="plus" title={lf("Add custom blocks?")} onClick={this.handleCustomBlocksClick} onKeyDown={this.handleButtonKeydown} /> : undefined}
                 {!meta.numErrors ? null : <span className='ui label red'>{meta.numErrors}</span>}
             </div>
+            {showGithub ? <GithubTreeItem parent={this.props.parent} githubId={mainPkg.header.githubId} /> : undefined}
             {show ? pxt.Util.concat(pkg.allEditorPkgs().map(p => this.filesWithHeader(p))) : undefined}
         </div>;
     }
 }
 
+interface GithubTreeItemProps extends ISettingsProps {
+    githubId: string;
+}
+
+interface GithubTreeItemState {
+    pushPulling?: boolean;
+}
+
+class GithubTreeItem extends sui.UIElement<GithubTreeItemProps, GithubTreeItemState> {
+    constructor(props: GithubTreeItemProps) {
+        super(props);
+        this.state = {};
+        this.handleClick = this.handleClick.bind(this);
+        this.handleButtonKeydown = this.handleButtonKeydown.bind(this);
+    }
+
+    private handleButtonKeydown(e: React.KeyboardEvent<HTMLElement>) {
+        e.stopPropagation();
+    }
+
+    private handleClick(e: React.MouseEvent<HTMLElement>) {
+        const { githubId } = this.props;
+        if (!githubId) {
+            this.createRepositoryAsync().done();
```

Comment:
Why `.done()`?

Reply:
Purely a style gimmic. It clearly state that no one is checking or waiting on this promise; an you cannot accidently return it.